crates/net/network/src/transactions.rs

```diff
@@ -1196,18 +1640,172 @@ impl TransactionFetcher {
+            // remove requested hashes from buffered hashes
+            debug_assert!(
+                || -> bool {
+                    for hash in &new_announced_hashes {
+                        if self.buffered_hashes.contains(hash) {
+                            return false
+                        }
+                    }
+                    true
+                }(),
+                "broken invariant `buffered-hashes` and `unknown-hashes`"
+            );
+
+            // stores a new request future for the request
+            self.inflight_requests.push(GetPooledTxRequestFut::new(
+                peer_id,
+                new_announced_hashes,
+                rx,
+            ))
+        }
+
+        None
+    }
+
+    /// Tries to fill request so that the respective tx response is at its size limit. It does so
+    /// by taking buffered hashes for which peer is listed as fallback peer. If this is an eth68
+    /// request, the accumulated size of transactions corresponding to parameter hashes, must also
+    /// be passed as parameter.
+    fn fill_request_for_peer(
+        &mut self,
+        hashes: &mut Vec<TxHash>,
+        peer_id: PeerId,
+        mut acc_eth68_size: Option<usize>,
+    ) {
+        debug_assert!(
+            acc_eth68_size.is_none() || {
+                let mut acc_size = 0;
+                for &hash in hashes.iter() {
+                    _ = self.include_eth68_hash(&mut acc_size, hash);
+                }
+                Some(acc_size) == acc_eth68_size
+            },
+            "broken invariant `acc-eth68-size` and `hashes`"
+        );
+
+        for hash in self.buffered_hashes.iter() {
+            // if this request is eth68 txns, check the size metadata
+            if let Some(acc_size_response) = acc_eth68_size.as_mut() {
+                let size_accumulated_eval = self.include_eth68_hash(acc_size_response, *hash);
+                match size_accumulated_eval {
+                    Some(true) => (),
+                    Some(false) | None => {
+                        trace!(
+                            target: "net::tx",
+                            peer_id=format!("{peer_id:#}"),
+                            hash=format!("{hash:#}"),
+                            size=self.eth68_meta.get(hash).expect("should find size in `eth68-meta`"),
+                            acc_size_response=acc_size_response,
+                            MAX_FULL_TRANSACTIONS_PACKET_SIZE=MAX_FULL_TRANSACTIONS_PACKET_SIZE,
+                            "found buffered hash for peer but can't fit it into request"
+                        );
+                        if let Some(false) = size_accumulated_eval {
+                            continue
+                        }
+                        if size_accumulated_eval.is_none() {
+                            // tx response is full
+                            break
+                        }
```

Comment:
this could be `if is_none() {} else {}` 

I guess we can also use individual match arms for None, Some(false) and the trace a closure. but not sure if I'd actually prefer that